src/tracks/playabletrack/wavetrack/ui/SpectrumView.cpp

```diff
@@ -608,13 +609,14 @@ void DrawClipSpectrum(TrackPanelDrawingContext &context,
 
    // Draw clip edges, as also in waveform view, which improves the appearance
    // of split views
-   params.DrawClipEdges( dc, rect );
+   params.DrawClipEdges( dc, rect, selected);
 }
 
 }
 
-void SpectrumView::DoDraw( TrackPanelDrawingContext &context,
-                                const WaveTrack *track,
+void SpectrumView::DoDraw(TrackPanelDrawingContext& context,
+                                const WaveTrack* track,
+                                const std::shared_ptr<WaveClip>& selectedClip,
```

Comment:
This might be a simple pointer.  It's safe to assume no clip you point to will be destroyed during the drawing of the track panel.  So ownership semantics are irrelevant to the function.